flink-connectors/flink-connector-gcp-pubsub/src/main/java/org/apache/flink/streaming/connectors/gcp/pubsub/table/PubSubConnectorConfigOptions.java

```diff
@@ -0,0 +1,27 @@
+package org.apache.flink.streaming.connectors.gcp.pubsub.table;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ConfigOptions;
+
+/**
+ * Options for PubSub tables supported by the {@code CREATE TABLE ... WITH ...} clause of the Flink
+ * SQL dialect and the Flink Table API.
+ */
+@PublicEvolving
+public class PubSubConnectorConfigOptions {
+
+    public static final ConfigOption<String> PROJECT_NAME =
+            ConfigOptions.key("projectName")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Name of the PubSub project backing this table.");
+
+    public static final ConfigOption<String> SUBSCRIPTION =
+            ConfigOptions.key("subscription")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Name of the PubSub subscription backing this table.");
+
+    public static final String IDENTIFIER = "pubsub";
```

Comment:
This isn't a config option. We currently have these reside in the factory class. I don't think that's ideal (since it is internal), but for now we should move this one to the factory as well, I think.